src/sentry/ddm/api.py

```diff
@@ -0,0 +1,386 @@
+        for totals_item in totals:
+            grouped_values = []
+            for group_by in group_bys or ():
+                grouped_values.append((group_by.key, totals_item.get(group_by.key)))
+
+            # The group key must be ordered, in order to be consistent across executions.
+            group_key = tuple(sorted(grouped_values))
+            group_metrics = intermediate_groups.setdefault(group_key, {})
+            metric_values = group_metrics.setdefault(metric_name, [[], 0])
+            # The item at position 1 is the "totals".
+            metric_values[1] = totals_item.get("aggregate_value")
+
+        meta = snuba_result["meta"]
+        for meta_item in meta:
+            meta_name = meta_item.get("name")
+            meta_type = meta_item.get("type")
+
+            # Since we have to handle multiple time series, we map the aggregate value to the actual
+            # metric name that was queried.
+            if meta_name == "aggregate_value":
+                intermediate_meta[metric_name] = meta_type
+            else:
+                intermediate_meta[meta_name] = meta_type
+
+    translated_groups = []
+    for group_key, group_metrics in sorted(intermediate_groups.items(), key=lambda v: v[0]):
+        # This case should never happen, since if we have start and intervals not None
+        assert start is not None and end is not None and intervals is not None
+
+        start_seconds = int(start.timestamp())
+        num_intervals = len(intervals)
+
+        series = {}
+        totals = {}
+        for metric_name, metric_values in sorted(group_metrics.items(), key=lambda v: v[0]):
+            series[metric_name] = _generate_full_series(
+                start_seconds, num_intervals, interval, metric_values[0]
+            )
+            totals[metric_name] = metric_values[1]
+
+        inner_group = {
+            "by": {name: value for name, value in group_key},
+            "series": series,
+            "totals": totals,
+        }
+
+        translated_groups.append(inner_group)
+
+    translated_meta = [
+        {"name": meta_name, "type": meta_type} for meta_name, meta_type in intermediate_meta.items()
+    ]
+
+    return {
+        "intervals": intervals,
+        "groups": translated_groups,
+        "meta": translated_meta,
+        "start": start,
+        "end": end,
+    }
+
+
+def _execute_series_and_totals_query(
+    organization: Organization, use_case_id: UseCaseID, interval: int, base_query: MetricsQuery
+) -> Mapping[str, Any]:
+    dataset = Dataset.Metrics if use_case_id == UseCaseID.SESSIONS else Dataset.PerformanceMetrics
+    request = Request(
+        dataset=dataset.value,
+        query=base_query,
+        app_id="default",
+        tenant_ids={"referrer": "metrics.data", "organization_id": organization.id},
+    )
+
+    base_query.rollup = Rollup(interval=interval, granularity=_get_granularity(interval))
+    series_result = run_query(request=request)
```

Comment:
Let's run a single query with `totals=True` -- as long as there are bugs, we can temporarly disable that but rather not workaround it. Please ask to prioritize the fix, instead.